risc0/zkvm/src/lib.rs

```diff
@@ -18,17 +18,23 @@
 #![deny(missing_docs)]
 
 extern crate alloc;
+mod fault_ids;
+pub use fault_ids::{FAULT_CHECKER_ELF, FAULT_CHECKER_ID};
 
 pub mod guest;
 #[cfg(not(target_os = "zkvm"))]
 mod host;
 pub mod serde;
 pub mod sha;
 
+#[cfg(feature = "std")]
+mod fault_monitor;
```

Comment:
We should avoid using "std" as a way to enable features I think.

Reply:
I could change this to `feature = "fault-proof"`.

Backstory: the reason this exists is because we have a no-std test and without this it broke the guest-methods compilation test for no-std. If I exposed this without this cfg, it broke the test. the fault_monitor mod depends on rrs-lib which depends on std :(

The fault monitor struct is used by both fault-checker and the host. So we only really need to expose this for the fault checker. Maybe having fault-proof feature doesn't need to be temporary but we could use it to toggle this? Exposing this in a normal guest context might not be very useful even if you have std enabled.

Reply:
I ended up creating a new feature called fault-proof and kept enable-fault proof. the `feature = "fault-proof"` will be used to expose the `fault_monitor` to the prover and `enable-fault-proof` will be used as a temporary feature to toggle the executor to run the fault checker.